searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_base.h

```diff
@@ -0,0 +1,72 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+#include "blockable_maintenance_job.h"
+#include "document_db_maintenance_config.h"
+#include "i_disk_mem_usage_listener.h"
+#include "iclusterstatechangedhandler.h"
+#include <vespa/searchlib/common/idocumentmetastore.h>
+
+namespace proton {
+    class IDiskMemUsageNotifier;
+    class IClusterStateChangedNotifier;
+    class IOperationStorer;
+    class ILidSpaceCompactionHandler;
+    class IDocumentScanIterator;
+    class RemoveOperationsRateTracker;
+}
+
+namespace proton {
+
+/**
+ * Job that regularly checks whether lid space compaction should be performed
```

Comment:
Could update comment to indicate that this is a base class.

Reply:
Ok, will follow up in separate PR

Reply:
Fixed